File: Solutions/1700/1700.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1700.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-10-19 10:38:35
***************************************************************************************************/

int countStudents(int* students, int studentsSize, int* sandwiches, int sandwichesSize){
    for (int i=0; i<sandwichesSize; i++) {
        int j=0;
        bool ok=false;
        for (; j<studentsSize; j++) {
            if (sandwiches[i] == students[j]) {
                students[j] = 2;
                ok = true;
                break;
            }
        }
        if (!ok) return studentsSize - i;
    }
    return 0;
}

File: Solutions/1700/1700.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/10/19 10:35
# File: 1700.py
# Desc: 


class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        s = collections.Counter(students)
        while sandwiches and sandwiches[0] in s:
            s[sandwiches[0]] -= 1
            if s[sandwiches[0]] == 0:
                del s[sandwiches[0]]
            sandwiches.pop(0)
        # print(s)
        if s:
            return list(s.values())[0]
        return 0
